File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int put = nums1.size() - 1;
    int scnd = nums2.size() - 1;
    int first = m - 1;

    while(first >= 0 && scnd >= 0){

        if(nums2[scnd] >= nums1[first]){
            nums1[put] = nums2[scnd];
            scnd--;
            put--;
        }

        else{
            nums1[put] = nums1[first];
            put--;
            first--;
        } 

    }

    while (scnd >= 0) {
        nums1[put--] = nums2[scnd--];
    }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        if (n == 0)
            return;

        int put = nums1.length - 1;
        int i = m - 1;
        int j = n - 1;

        while (i >= 0 && j >= 0) {
            if (nums2[j] >= nums1[i]) {
                nums1[put] = nums2[j];
                put--;
                j--;
            } else {
                nums1[put] = nums1[i];
                put--;
                i--;
            }
        }

        while (j >= 0) {
            nums1[put] = nums2[j];
            j--;
            put--;
        }

    }
}